# 단일표본 t검정

## Example 1 : 단일표본 t검정
100명의 키 정보가 들어 있는 데이터가 있다.데이터가 정규성을 만족하는지 확인하라.
그리고 평균키는 165라 판단할수 있는지 귀무가설과 대립가설을 설정한 후 유의수준 5%로 검정하라.

[데이터 마님 scipy stats 모듈 - 단일표본 t검정](https://www.datamanim.com/dataset/97_scipy/scipy.html#t-one-sample)

In [8]:
import pandas as pd 
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/scipy/height1.csv')
df.head()

height
0  160.237691
1  164.747324
2  165.401628
3  168.801627
4  153.199021

In [9]:
# 정규성 검정 : shapiro
from scipy.stats import shapiro
shapiro(df["height"])

ShapiroResult(statistic=0.9872670769691467, pvalue=0.455832302570343)

- p-value가 약 0.46으로, 유의수준 0.05보다 크다. 따라서 귀무가설을 기각하지 않는다. 정규성을 따른다고 할 수 있다. 

In [11]:
# 단일표본 t검정
from scipy.stats import ttest_1samp
ttest_1samp(df["height"], 165)

Ttest_1sampResult(statistic=3.2017884987150644, pvalue=0.0018367171548080209)

- p-value가 약 0.002로, 유의수준 0.05보다 작다. 따라서 귀무가설을 기각한다. 평균 키는 165라고 할 수 없다.

## Example 2 : 윌콕슨 순위 부호 검정 (비모수 검정)
100명의 키 정보가 들어 있는 데이터가 있다.데이터가 정규성을 만족하는지 확인하라.
그리고 평균키는 165라 판단할수 있는지 귀무가설과 대립가설을 설정한 후 유의수준 5%로 검정하라.

[데이터 마님 scipy stats 모듈 - 단일표본 t검정](https://www.datamanim.com/dataset/97_scipy/scipy.html#t-one-sample)

In [5]:
import pandas as pd 
df = pd.read_csv('https://raw.githubusercontent.com/Datamanim/datarepo/main/scipy/height2.csv')
df.head()

height
0  182.012475
1  166.857081
2  159.633265
3  159.633265
4  169.428478

In [6]:
# 정규성 검정 : shapiro
from scipy.stats import shapiro
shapiro(df["height"])

ShapiroResult(statistic=0.9672006368637085, pvalue=0.013552471995353699)

- p-value가 0.013으로, 유의수준 0.05보다 작다. 따라서 귀무가설을 기각한다. 정규성을 따르지 않는다고 할 수 있다.

### 정규성을 따르지 않는 단일표본 검정은 '일표본 윌콕슨 부호검정'을 이용 

In [7]:
from scipy.stats import wilcoxon
wilcoxon(df["height"]-165)

WilcoxonResult(statistic=1952.0, pvalue=0.04880534424649582)

- p-value가 0.049로, 유의수준 0.05보다 작다. 따라서 귀무가설을 기각한다. 평균키는 165라고 할 수 없다.